### DISCUSSION TITLE
compute capability 89 problem

### DISCUSSION THREAD

**Discussion by keebborg** [TinkerTools/tinker-gpu#218] [Q&A] [answered]
Hello!
I have a problem to run Tinker9 after installation:

tinker9 dynamic -k dhfr2.key dhfr2.xyz 1000 1.0 1.0 4 298 1.0
...
Terminating with uncaught exception :  The source code should be updated for compute capability 89; Please let us know at ../tinker9/src/cudart/gpucard.cpp:149

Maybe someone knows how to solve this problem? I use RTX 4090, cuda12 and Nvidia Driver Version: 525.60.13.

Thanks in advance

**Comment by pren** [comment]
Make sure your hpc_sdk matches cuda 12.

I have no issue using cuda_11.2 on 4090.



From: keebborg ***@***.***>
Date: Tuesday, March 14, 2023 at 2:12 PM
To: TinkerTools/tinker9 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TinkerTools/tinker9] compute capability 89 problem (Discussion #218)

Hello!
I have a problem to run Tinker9 after installation:

tinker9 dynamic -k dhfr2.key dhfr2.xyz 1000 1.0 1.0 4 298 1.0
...
Terminating with uncaught exception : The source code should be updated for compute capability 89; Please let us know at ../tinker9/src/cudart/gpucard.cpp:149

Maybe someone knows how to solve this problem? I use RTX 4090, cuda12 and Nvidia Driver Version: 525.60.13.

Thanks in advance

**Comment by zhi-wang** [comment, marked as answer]
Where/When/How did you get/compile the Tinker9 executable (from)? I believe the source code of your executable is older than the current version. Code to support cc89 was added about two months ago. If a newer cc is still missing, this error message would have been thrown from Line 154 but in your error message, the error was thrown from Line 149.

**Reply by keebborg** [reply]
Dear Zhi Wang, 

Indeed, the problem was related to an older version of Tinker9 (was older than two months). Now it works without errors!

Thanks!
